ts-client/src/dlmm/index.ts

```diff
@@ -3388,14 +3390,20 @@ export class DLMM {
       this.lbPair.binStep
     );
 
+    // Extra binArrays that doesn't exist in the binArraysForSwap
+    const extraBinArrays = binArrays.map(item => item.publicKey).filter(binArrayPubkey => !binArraysForSwap.has(binArrayPubkey));
+    const binArraysForSwapKeys = Array.from(binArraysForSwap.keys());
+    const extraBinArrayIndexEnd = maxExtraBinArrays | 0;
+    const binArraysPubkey = [...binArraysForSwapKeys, ...extraBinArrays.slice(0, extraBinArrayIndexEnd)];
```

Comment:
better to refactor: 
const extraBinArray = maxExtraBinArrays | 0;
if extraBinArray ! = 0 {
   other codes
}

Reply:
actually the `slice` method can handle many cases for `extraBinArrayIndexEnd`, if
- extraBinArrayIndexEnd = 0 then return empty array.
- extraBinArrayIndexEnd > extraBinArrays.length then return all items in array.

Perhap we add an validation to check for maxExtraBinArrays >= 0

Reply:
I mean if you add the validation before line 3393, we should save some resource to filter, etc
 if (extraBinArrayIndexEnd < 0) { throw error}
 if (extraBinArrayIndexEnd > 0) {do some work}

Reply:
updated the logic